# Lab 8 Exercise 7

## Method with parameter and return value

1. สร้าง console application project

```
    dotnet new console --name Lab08_Ex07
```

2. เปลี่ยน code ให้เป็นดังต่อไปนี้

> ```csharp
>  1. var adder = new MyProgram();
>  2. int x = 2, y = 3;
>  3. int add_result = adder.add(x, y);
>  4. System.Console.WriteLine($"{x} + {y} = {add_result}");
>  5. 
>  6. int i = 2, j = 3;
>  7. int multiply_result = adder.multiply(i, j);
>  8. System.Console.WriteLine($"{i} x {j} = {multiply_result}");
>  9. 
> 10. class MyProgram
> 11. {
> 12.     public int add(int a, int b)
> 13.     {
> 14.         return a + b;
> 15.     }
> 16.     public int multiply(int a, int b)
> 17.     {
> 18.         return a * b;
> 19.     }
> 20. }
> ```



3. Build project โดยการใช้คำสั่ง

```
    dotnet build  Lab08_Ex07
```

4. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 3



*คำตอบ*

PS C:\Users\user\Downloads\Lab\Lab08-CSharp-Class-Part-I> dotnet build  Lab08_Ex07
Restore complete (0.2s)
  Lab08_Ex07 succeeded (2.1s) → Lab08_Ex07\bin\Debug\net9.0\Lab08_Ex07.dll

Build succeeded in 2.7s

5. Run project โดยการใช้คำสั่ง

```
    dotnet run --project Lab08_Ex07
```

6. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 5





*คำตอบ*

PS C:\Users\user\Downloads\Lab\Lab08-CSharp-Class-Part-I> dotnet run --project Lab08_Ex07

2 + 3 = 5

2 x 3 = 6

7. อธิบายสิ่งที่พบในการทดลอง



*คำตอบ*

ไม่พบ error หรือ warning เเละโค้ดนี้มีการสร้างอ็อบเจ็กต์ adder จากคลาส MyProgram, แล้วเรียกใช้เมธอด add และ multiply

เพื่อคำนวณผลบวกและผลคูณของตัวเลขสองตัวที่ถูกส่งเข้าไปในแต่ละเมธอด.

ผลลัพธ์จากการคำนวณจะแสดงบนคอนโซลตามลำดับที่กำหนดไว้ในคำสั่ง Console.WriteLine

8. นำ code ในข้อ 2 มารันบน polyglot notebook  โดยใส่ใน  code cell ด้านล่างนี้


In [ ]:
// พื้นที่เขียน code ตามข้อ 2
var adder = new MyProgram();
   int x = 2, y = 3;
   int add_result = adder.add(x, y);
   System.Console.WriteLine($"{x} + {y} = {add_result}");
   
   int i = 2, j = 3;
   int multiply_result = adder.multiply(i, j);
   System.Console.WriteLine($"{i} x {j} = {multiply_result}");
   
  class MyProgram
  {
      public int add(int a, int b)
      {
          return a + b;
      }
      public int multiply(int a, int b)
      {
          return a * b;
      }
  }

2 + 3 = 5
2 x 3 = 6


8.1 ผลการรันเป็นอย่างไร



*คำตอบ*

2 + 3 = 5

2 x 3 = 6

8.2 เหมือนหรือต่างจาก console application อย่างไร

*คำตอบ*

เหมือนกัน

## คำถาม

1. ให้เขียน method ที่รับ parameter เป็น string และส่งกลับค่าความยาวของ string นั้น

In [ ]:
// พื้นที่เขียน code และทดสอบ
var program = new MyProgram();

string text = "Box";
int length = program.GetLength(text);
System.Console.WriteLine($"ความยาวของ \"{text}\" คือ {length}");

class MyProgram
{
    public int GetLength(string str)
    {
        return str.Length;
    }
}

ความยาวของ "Box" คือ 3


*คำตอบ*

ความยาวของ "Box" คือ 3

2. ให้สร้างโปรแกรมสูตรคูณ โดยประยุกต์ใช้เมธอด  `public int multiply(int a, int b)`


In [ ]:
// พื้นที่เขียน code และทดสอบ
 class MyProgram
{
    // เมธอดที่คูณตัวเลข 2 ตัวและส่งกลับผลลัพธ์
    public int Multiply(int a, int b)
    {
        return a * b;  // คำนวณผลคูณ
    }
}

   var calculator = new MyProgram();

        // กำหนดตัวเลขที่ต้องการให้คูณ (เช่น ตารางสูตรคูณ 2)
        int number = 9;

        // แสดงผลตารางสูตรคูณจาก 1 ถึง 12
        for (int i = 1; i <= 12; i++)
        {
            int result = calculator.Multiply(number, i); // เรียกเมธอด multiply
            Console.WriteLine($"{number} x {i} = {result}");
        }

9 x 1 = 9
9 x 2 = 18
9 x 3 = 27
9 x 4 = 36
9 x 5 = 45
9 x 6 = 54
9 x 7 = 63
9 x 8 = 72
9 x 9 = 81
9 x 10 = 90
9 x 11 = 99
9 x 12 = 108


*คำตอบ*

9 x 1 = 9

9 x 2 = 18

9 x 3 = 27

9 x 4 = 36

9 x 5 = 45

9 x 6 = 54

9 x 7 = 63

9 x 8 = 72

9 x 9 = 81

9 x 10 = 90

9 x 11 = 99

9 x 12 = 108

3. จากโปรแกรมในคำถามข้อที่ 2 ให้สร้างเป็นสูตรคูณ แม่ 2 - 12

In [ ]:
// พื้นที่เขียน code และทดสอบ
// พื้นที่เขียน code และทดสอบ
var program = new MyProgram();

for (int i = 1; i <= 12; i++) // วนลูปคูณ 1 - 12
{
    for (int num = 2; num <= 12; num++) // เปลี่ยนตัวเลขแม่สูตรคูณ (2 - 5)
    {
        int result = program.multiply(num, i);
        System.Console.Write($"{num} × {i} = {result}\t"); // ใช้ \t เพื่อจัดตำแหน่งให้อยู่ในคอลัมน์เดียวกัน
    }
    System.Console.WriteLine(); // ขึ้นบรรทัดใหม่เมื่อครบทุกแม่
}

class MyProgram
{
    public int multiply(int a, int b)
    {
        return a * b;
    }
}


2 × 1 = 2	3 × 1 = 3	4 × 1 = 4	5 × 1 = 5	6 × 1 = 6	7 × 1 = 7	8 × 1 = 8	9 × 1 = 9	10 × 1 = 10	11 × 1 = 11	12 × 1 = 12	
2 × 2 = 4	3 × 2 = 6	4 × 2 = 8	5 × 2 = 10	6 × 2 = 12	7 × 2 = 14	8 × 2 = 16	9 × 2 = 18	10 × 2 = 20	11 × 2 = 22	12 × 2 = 24	
2 × 3 = 6	3 × 3 = 9	4 × 3 = 12	5 × 3 = 15	6 × 3 = 18	7 × 3 = 21	8 × 3 = 24	9 × 3 = 27	10 × 3 = 30	11 × 3 = 33	12 × 3 = 36	
2 × 4 = 8	3 × 4 = 12	4 × 4 = 16	5 × 4 = 20	6 × 4 = 24	7 × 4 = 28	8 × 4 = 32	9 × 4 = 36	10 × 4 = 40	11 × 4 = 44	12 × 4 = 48	
2 × 5 = 10	3 × 5 = 15	4 × 5 = 20	5 × 5 = 25	6 × 5 = 30	7 × 5 = 35	8 × 5 = 40	9 × 5 = 45	10 × 5 = 50	11 × 5 = 55	12 × 5 = 60	
2 × 6 = 12	3 × 6 = 18	4 × 6 = 24	5 × 6 = 30	6 × 6 = 36	7 × 6 = 42	8 × 6 = 48	9 × 6 = 54	10 × 6 = 60	11 × 6 = 66	12 × 6 = 72	
2 × 7 = 14	3 × 7 = 21	4 × 7 = 28	5 × 7 = 35	6 × 7 = 42	7 × 7 = 49	8 × 7 = 56	9 × 7 = 63	10 × 7 = 70	11 × 7 = 77	12 × 7 = 84	
2 × 8 = 16	3 × 8 = 24	4 × 8 = 32	5 × 8 = 40	6 × 8 = 48	7 × 8 = 56	8 × 8 = 64	9 × 8 = 72	10 × 8 = 80	11 × 8 = 88	12 × 8 = 96	
2 × 9 = 18	3 ×

*คำตอบ*

2 × 1 = 2	3 × 1 = 3	4 × 1 = 4	5 × 1 = 5	6 × 1 = 6	7 × 1 = 7	8 × 1 = 8	9 × 1 = 9	10 × 1 = 10	11 × 1 = 11	12 × 1 = 12	

2 × 2 = 4	3 × 2 = 6	4 × 2 = 8	5 × 2 = 10	6 × 2 = 12	7 × 2 = 14	8 × 2 = 16	9 × 2 = 18	10 × 2 = 20	11 × 2 = 22	12 × 2 = 24	

2 × 3 = 6	3 × 3 = 9	4 × 3 = 12	5 × 3 = 15	6 × 3 = 18	7 × 3 = 21	8 × 3 = 24	9 × 3 = 27	10 × 3 = 30	11 × 3 = 33	12 × 3 = 36	

2 × 4 = 8	3 × 4 = 12	4 × 4 = 16	5 × 4 = 20	6 × 4 = 24	7 × 4 = 28	8 × 4 = 32	9 × 4 = 36	10 × 4 = 40	11 × 4 = 44	12 × 4 = 48	

2 × 5 = 10	3 × 5 = 15	4 × 5 = 20	5 × 5 = 25	6 × 5 = 30	7 × 5 = 35	8 × 5 = 40	9 × 5 = 45	10 × 5 = 50	11 × 5 = 55	12 × 5 = 60	

2 × 6 = 12	3 × 6 = 18	4 × 6 = 24	5 × 6 = 30	6 × 6 = 36	7 × 6 = 42	8 × 6 = 48	9 × 6 = 54	10 × 6 = 60	11 × 6 = 66	12 × 6 = 72	

2 × 7 = 14	3 × 7 = 21	4 × 7 = 28	5 × 7 = 35	6 × 7 = 42	7 × 7 = 49	8 × 7 = 56	9 × 7 = 63	10 × 7 = 70	11 × 7 = 77	12 × 7 = 84	

2 × 8 = 16	3 × 8 = 24	4 × 8 = 32	5 × 8 = 40	6 × 8 = 48	7 × 8 = 56	8 × 8 = 64	9 × 8 = 72	10 × 8 = 80	11 × 8 = 88	12 × 8 = 96	

2 × 9 = 18	3 × 9 = 27	4 × 9 = 36	5 × 9 = 45	6 × 9 = 54	7 × 9 = 63	8 × 9 = 72	9 × 9 = 81	10 × 9 = 90	11 × 9 = 99	12 × 9 = 108	

2 × 10 = 20	3 × 10 = 30	4 × 10 = 40	5 × 10 = 50	6 × 10 = 60	7 × 10 = 70	8 × 10 = 80	9 × 10 = 90	10 × 10 = 100	11 × 10 = 110	12 × 10 = 120	

2 × 11 = 22	3 × 11 = 33	4 × 11 = 44	5 × 11 = 55	6 × 11 = 66	7 × 11 = 77	8 × 11 = 88	9 × 11 = 99	10 × 11 = 110	11 × 11 = 121	12 × 11 = 132	

2 × 12 = 24	3 × 12 = 36	4 × 12 = 48	5 × 12 = 60	6 × 12 = 72	7 × 12 = 84	8 × 12 = 96	9 × 12 = 108	10 × 12 = 120	11 × 12 = 132	12 × 12 = 144	